src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -132,11 +140,18 @@ async function put_bucket_lifecycle(req) {
         }
 
         if (rule.Expiration?.length === 1) {
+            if (rule.Expiration.length > 1) {
+                throw new S3Error(S3Error.MalformedXML);
+            }
             current_rule.expiration = parse_expiration(rule.Expiration[0]);
             reject_empty_field(current_rule.expiration);
         }
 
         if (rule.AbortIncompleteMultipartUpload?.length === 1) {
+            if (rule.Filter?.Tag || rule.Filter?.ObjectSizeGreaterThan || rule.Filter?.ObjectSizeLessThan) {
+                dbg.error('Rule should not include AbortIncompleteMultipartUpload with Tags or ObjectSize filters', rule);
+                throw new S3Error({ ...S3Error.InvalidArgument, message: 'AbortIncompleteMultipartUpload cannot be specified with Tags' });
```

Comment:
probably need different error message for ObjectSizeGreaterThan and ObjectSizeLessThan

Reply:
Updated, thanks